avaframe/com1DFAPy/com1DFAtools.py

```diff
@@ -298,6 +323,160 @@ def computeTimeStep(cfg, particles, fields, dem, Ment, Cres, Tcpu):
     return particles, fields, Tcpu
 
 
+def computeLeapFrogTimeStep(cfg, particles, fields, dt, dem, Ment, Cres, Tcpu):
+    """ perform all computations that belong to one time step """
+
+    log.info('Use LeapFrog time stepping')
+    # start timing
+    startTime = time.time()
+    tcpuForce = time.time() - startTime
+    Tcpu['Force'] = Tcpu['Force'] + tcpuForce
+
+    # dtK5 is half time step
+    dtK5 = 0.5 * dt
+    log.info('dt used now is %f' % dt)
+
+    # load required DEM and mesh info
+    csz = dem['header'].cellsize
+    Nx = dem['Nx']
+    Ny = dem['Ny']
+    Nz = dem['Nz']
+
+    # particle properties
+    mass = particles['m']
+    xK = particles['x']
+    yK = particles['y']
+    zK = particles['z']
+    uxK = particles['ux']
+    uyK = particles['uy']
+    uzK = particles['uz']
+
+    #+++++++++++++Time integration using leapfrog 'Drift-Kick-Drif' scheme+++++
+    # first predict position at time t_(k+0.5)
+    # 'DRIFT'
+    xK5 = xK + dt * 0.5 * uxK
+    yK5 = yK + dt * 0.5 * uyK
+    zK5 = zK + dt * 0.5 * uzK
+    # update position from particles
+    particles['x'] = xK5
+    particles['y'] = yK5
+    # For now z-position is taken from DEM - no detachment enforces...
+    particles, _ = geoTrans.projectOnRasterVect(dem, particles, interp='bilinear')
+    # TODO: do we need to update also h from particles?? I think yes! also mass, ent, res
+    ##particles['h'] = ?
+
+    # 'KICK'
+    # compute velocity at t_(k+0.5)
+    # first compute force at t_(k+0.5)
+    force = computeForceVect(cfg, particles, dem, Ment, Cres, dtK5)
+    particles, force = computeForceSPH(cfg, particles, force, dem)
+    mass = particles['m']
+    uxNew = uxK + (force['forceX'] + force['forceSPHX']) * dt / mass
+    uyNew = uyK + (force['forceY'] + force['forceSPHY']) * dt  / mass
+    uzNew = uzK + (force['forceZ'] + force['forceSPHZ']) * dt / mass
+
+    # 'DRIF'
+    # now update position at t_(k+ 1)
+    xNew = xK5 + dtK5 * uxNew
+    yNew = yK5 + dtK5 * uyNew
+    zNew = zK5 + dtK5 * uzNew
```

Comment:
you use the estimated position xK5 as starting point, is that correct? (I am just asking, I have no Idea how it works)